GitUI/Theming/ThemeController.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Reflection;
+using System.Windows.Forms;
+using GitCommands;
+using GitExtUtils.GitUI.Theming;
+
+namespace GitUI.Theming
+{
+    public class ThemeController : Theme
```

Comment:
Why do we need to inherit from `Theme`?

Reply:
because it provides the same interface to get Application and System colors, to be used in `ColorHelper`.

Reply:
I'm finding it confusing and prone to a number of problems including maintainability and concurrency... 😕 


We are mixing state and behaviours and this, in my experience, often leads to tighter coupling, lower cohesion, inability to inject/substitute behaviours, serialise/persist state, and threading issues.
When I think of a theme, I think of a set of properties (e.g. colours, images, etc.) that are somehow persisted. When I think of a theme manager/controller I can imagine a behavioural system that loads/saves themes and otherwise allows interacting with a theme.

By separating state and behaviour we will be able to alter theme definitions and support multiple versions, and/or substitute how themes are loaded or persisted (e.g. in a flat or a json file, read Widnows theme definitions etc) with minimal overhead.

Granted in this specific case we are dealing with themes in two points - at the app load and the settings dialog, so the risk of concurrency is likely low, we are increasing our tech debt surface and may be giving wrong ideas to other contributors.

Reply:
`ThemeManager` stores in-memory .Net `SystemColors` colors overrides to be accessed by `Win32ThemeHooks` hook overrides. Notice that `Win32ThemeHooks` only sees base class `Theme`'s interface, which is just a storage for color values definitions.

Instead of inheriting from `Theme`, `ThemeManager` could be refactored to include a read-write-able instance of `Theme`, which would be a sort of decoupling, though in my view purely cosmetic and not useful. Still if you disagree or have different perspective, I am 100% ok to change.

In contrast with `SystemColors`, GitExtensions app-specific color overrides are stored and persisted by already present `AppSettings` mechanism, from which the rest of application reads `AppColors` values. Changing it looks pretty expensive.

Currently `ThemeManager` provides a uniform interface to read and modify both `SystemColors` and `AppColors` which is helpful in `ThemeEditor`.

Reply:
Ok let's keep it as is.